### ISSUE TITLE
Problem with GTDB database  at add_names step

### ISSUE THREAD

**Issue by Kurtj-hub** [MGXlab/CAT_pack#71]
Thank you for adding support for the GTDB database via the new download command. Though I am running into an error (which is probably a simple fix), but rather than me monkeying around with the code any more than I already have. If you have time and could look at the following error and attached names.dmp file or provide any other guidance for troubleshooting that would be greatly appreciated. 

[names.txt](https://github.com/dutilh/CAT/files/8393501/names.txt)

When I run the following (which worked with the nr database, wildcards are relevant place holders)
Note: same version of diamond (2.0.14) was used for all steps as far as I am aware. 
- database and taxonomy point to the relevant GTDB folders. 

`CAT_pack/CAT contigs -c {input.contigs} -p {input.faa} --force -n {threads} -d {DBDIR}/CAT_database -t {DBDIR}/CAT_taxonomy --out_prefix {wildcards.sample}/CAT/{wildcards.sample}` 

`CAT_pack/CAT add_names -i {wildcards.sample}/CAT/{wildcards.sample}.contig2classification.txt -o {wildcards.sample}/CAT/{wildcards.sample}.contig2classification.official_names.txt -t {DBDIR}/CAT_taxonomy --only_official --force`
        
I get the following error....

`[2022-03-31 17:52:09] Loading file ~/databases/CAT_taxonomy/names.dmp.
Traceback (most recent call last):
  File "\~/CAT_pack/CAT", line 88, in <module>
    main()
  File "\~/CAT_pack/CAT", line 74, in main
    add_names.run()
  File "\~/CAT_pack/add_names.py", line 74, in run
    taxid2name = tax.import_names(args.names_dmp, args.log_file, args.quiet)
  File "\~/tax.py", line 39, in import_names
    if line[6] == 'scientific name':

IndexError: list index out of range` 

I expected this to complete and to get the following file with names.
`{wildcards.sample}/CAT/{wildcards.sample}.contig2classification.official_names.txt output.` 

Thanks!

**Comment by Kurtj-hub** [comment]
I missed that diamond is finishing abnormally above the quoted error which may in fact be the root error.

**Comment by gregorysprenger** [comment]
Bumping this as I'm running into the same issue with the latest version (v5.3). @Kurtj-hub, were you able to figure out the problem?

**Comment by samche42** [comment]
Hi there!

I've just bumped into the same problem. CAT runs flawlessly with the nr data but GTDB is tripping it up. I downloaded and decompressed the pre-prepared GTDB database (20231120_CAT_gtdb.tar.gz). 

Contents of std err:
```
Traceback (most recent call last):
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/bin/CAT", line 84, in <module>
    main()
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/bin/CAT", line 64, in main
    contigs.run()
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/share/cat-5.2.3-1/CAT_pack/contigs.py", line 310, in run
    lineage = tax.star_lineage(
              ^^^^^^^^^^^^^^^^^
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/share/cat-5.2.3-1/CAT_pack/tax.py", line 147, in star_lineage
    questionable_taxids = find_questionable_taxids(lineage,
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/share/cat-5.2.3-1/CAT_pack/tax.py", line 139, in find_questionable_taxids
    taxid_parent = lineage[i + 1]
                   ~~~~~~~^^^^^^^
IndexError: list index out of range
Traceback (most recent call last):
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/bin/CAT", line 84, in <module>
    main()
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/bin/CAT", line 70, in main
    add_names.run()
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/share/cat-5.2.3-1/CAT_pack/add_names.py", line 74, in run
    taxid2name = tax.import_names(args.names_dmp, args.log_file, args.quiet)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/nasapps/development/python/miniconda3/3.10/envs/CAT/share/cat-5.2.3-1/CAT_pack/tax.py", line 39, in import_names
    if line[6] == 'scientific name':
       ~~~~^^^
IndexError: list index out of range
```

**Comment by bastiaanvonmeijenfeldt** [comment]
Hi all,

I'm sorry for the late reply! This bug has now been fixed in #115. The solution involves downloading the latest code (not released as of yet) but also remaking the GTDB database files. With the release of v6.0.2 we will post those database files on tbb.bio.uu.nl/tina/CAT_pack_prepare/](https://tbb.bio.uu.nl/tina/CAT_pack_prepare/.

I hope this helps, at least in the very near future. :) I'm closing this issue for now, let us know if you walk into other problems, we're currently in active development again and ironing out bugs (@thauptfeld).

Best wishes,

Bastiaan

**Comment by samche42** [comment]
Thanks Bastiaan! CAT is an awesome tool - looking forward to the new release :)
